### PR TITLE
Implemented linspace frontend for paddle.tensor.creation

### LINKED ISSUE #18148: linspace

### PR DESCRIPTION
**Pull request by KevinUli**
Close #18148

### CODE REVIEW

**Comment by KevinUli** [comment]
Rounding .5 works differently for paddle as opposed to other frameworks. Paddle rounds .5 to the nearest even integer (ex: 2.5 -> 2, 3.5 -> 4), while other frameworks rounds .5 to the nearest largest absolute number (ex: -0.5 -> -1, 2.5 -> 3). I left it as failing tests for now.

**Comment by xoiga123** [comment]
> Rounding .5 works differently for paddle as opposed to other frameworks. Paddle rounds .5 to the nearest even integer (ex: 2.5 -> 2, 3.5 -> 4), while other frameworks rounds .5 to the nearest largest absolute number (ex: -0.5 -> -1, 2.5 -> 3). I left it as failing tests for now.

Can you write up a minimal snippet about this behavior? Is this for `linspace` only or with rounding in general? Because I get the usual behavior with `paddle.round`:
```python
>>> a = paddle.to_tensor([2.5])
>>> paddle.round(a)
Tensor(shape=[1], dtype=float32, place=Place(cpu), stop_gradient=True,
       [3.])
```

If it's only with `linspace`, then you'll have to do some special handling in the frontend. Eh is the `linspace` backend test for paddle passing?

**Comment by KevinUli** [comment]
On further inspection, it seems rounding works oddly for some frameworks.
Paddle and PyTorch rounds equivalently (though I can't seem to find out how they round), Numpy and Jax rounds by flooring, while Tensorflow rounds to lowest absolute integer. I currently set it such that it rounds to the lowest absolute integer as it is the closest behavior to paddle I can think of (Which works except for a few edge cases such that -1.125 rounds to 0 and 1.125 rounds to 0).
Paddle:
```
>>> start = paddle.to_tensor(-1.5, dtype="float32")
>>> stop = paddle.to_tensor(1.5, dtype="float64")
>>> paddle.linspace(start, stop, num=9, dtype="float64")
Tensor(shape=[9], dtype=float64, place=Place(cpu), stop_gradient=True,
       [-1.50000000, -1.12500000, -0.75000000, -0.37500000,  0.        ,
         0.37500000,  0.75000000,  1.12500000,  1.50000000])
>>> paddle.linspace(start, stop, num=9, dtype="int32")
Tensor(shape=[9], dtype=int32, place=Place(cpu), stop_gradient=True,
       [-1,  0,  0,  0,  0,  0,  0,  0,  1])
```
Torch:
```
>>> start = torch.as_tensor(-1.5, dtype=torch.float32)
>>> stop = torch.as_tensor(1.5, dtype=torch.float64)
>>> torch.linspace(start, stop, steps=9, dtype=torch.float64)
tensor([-1.5000, -1.1250, -0.7500, -0.3750,  0.0000,  0.3750,  0.7500,  1.1250, 1.5000], dtype=torch.float64)
>>> torch.linspace(start, stop, steps=9, dtype=torch.int32)
tensor([-1,  0,  0,  0,  0,  0,  0,  0,  1], dtype=torch.int32)
```
Numpy:
```
>>> start = np.asarray(-1.5, dtype="float32")
>>> stop = np.asarray(1.5, dtype="float64")
>>> np.linspace(start, stop, num=9, dtype="float64")
array([-1.5  , -1.125, -0.75 , -0.375,  0.   ,  0.375,  0.75 ,  1.125, 1.5  ])
>>> np.linspace(start, stop, num=9, dtype="int32")
array([-2, -2, -1, -1,  0,  0,  0,  1,  1])
```
Jax:
```
>>> start = jax.numpy.asarray(-1.5, dtype="float32")
>>> end = jax.numpy.asarray(1.5, dtype="float64")
>>> jax.numpy.linspace(start, end, num=9, dtype="float64")
Array([-1.5  , -1.125, -0.75 , -0.375,  0.   ,  0.375,  0.75 ,  1.125,
        1.5  ], dtype=float32)
>>> jax.numpy.linspace(start, end, num=9, dtype="int32")
Array([-2, -2, -1, -1,  0,  0,  0,  1,  1], dtype=int32)
```
Tensorflow:
```
>>> start = tf.convert_to_tensor(-1.5, dtype=tf.float32)
>>> end = tf.convert_to_tensor(1.5, dtype=tf.float64)
>>> ret = tf.linspace(start, end, num=9)
>>> ret
<tf.Tensor: shape=(9,), dtype=float32, numpy=
array([-1.5  , -1.125, -0.75 , -0.375,  0.   ,  0.375,  0.75 ,  1.125,
        1.5  ], dtype=float32)>
>>> tf.cast(ret, tf.int32)
<tf.Tensor: shape=(9,), dtype=int32, numpy=array([-1, -1,  0,  0,  0,  0,  0,  1,  1])>
```

**Comment by xoiga123** [comment]
Good point, currently `test_linspace` and all the frontend variations will fail if we use `dtype=helpers.get_dtypes("numeric")` instead of `"float"`. We'll need to modify the backends to account for this behaviour. Do you mind diving deep into this? I assume that the frameworks doesn't handle `dtype` in `linspace` in the same way. For example:

- [Paddle](https://github.com/PaddlePaddle/Paddle/blob/7ed818f83adf90404dd788baecea21d007aa1a80/python/paddle/tensor/creation.py#L306-L314): try to convert the arguments into `dtype` first, then split.
- [Numpy](https://github.com/numpy/numpy/blob/db4f43983cb938f12c311e1f5b7165e270c393b4/numpy/core/function_base.py#L176-L182): split first, then try to convert the result array into `dtype`.

As for which behaviour would be "more correct", I think we'll have to go with whatever is more mainstream.

**Comment by ivy-seed** [comment]
This PR has been labelled as stale because it has been inactive for more than 7 days. If you would like to continue working on this PR, then please add another comment or this PR will be closed in 7 days.

**Comment by ivy-seed** [comment]
This PR has been closed because it has been marked as stale for more than 7 days with no activity.
